docs/development/openocd_gdb_debugging.md

```diff
@@ -37,13 +42,131 @@ If you do not have vscode yet, the easiest way to install it on Ubuntu is via sn
     sudo snap install --classic code
 
 
+
+#### Windows (Ubuntu in WSL)
+
+##### Install WSL
+
+Configure your Machine for WSL and install a Ubuntu Distrobution. Instructions can be found here: https://learn.microsoft.com/en-us/windows/wsl/install . It has been tested on Windows 10 and 11 with Ubuntu 20.04.
+
+##### Install the GCC ARM embedded Toolchain
```

Comment:
i..don't know anymore tbh. For some reason it didn't work so i searched for other methods, i did a quick and dirty delete of embedded and just ran `sudo apt-get install -y gcc-arm-none-eabi` again. It seems to work fine for now. Let me make sure that it works on a clean install and I'll get back to you later. That would simplify the instructions for WSL a lot 👍

Reply:
Sounds good. One possible trick is that the vsCode extension expects gdb to be called `arm-none-eabi-gdb` and I did not find any way to configure it so the linux instruction includes making a link so that the extension finds gdb: https://github.com/bitcraze/crazyflie-firmware/blob/c630c82c7afc445080a994c9e3409a036b9c858d/docs/development/openocd_gdb_debugging.md?plain=1#L33

Reply:
it works, just by running `sudo apt-get install -y gcc-arm-none-eabi` alone. and there is another thing i want to add, which is that you have to tell VSCode to not update Extensions automatically.
The only option i found is to add this
```
{
  "extensions.autoCheckUpdates": false,
  "extensions.autoUpdate": false
}
```
to .vscode/settings.json, but i haven't tested it over two days, normally vscode doesn't update extensions immediately.

How can I commit to this PR? I'm fairly new to github, can I just commit to my branch in my fork or do i have to do anything else?

Reply:
Yes that is the way, Your branch called 'new_branch' is the one linked to this PR, so if you commit to that one and push it to your own repo, it will automatically update this PR as well

Reply:
Was busy on another project, but here is the finished PR. that `extensions.autoUpdate: false` is not required since you can just check the `ignore updates` Option in the Extension settings. I included it in the instructions.

I just got done fully validating the whole instructions for Debugging on WSL on a fresh everything windows pc. (since i broke my entire wsl system with a ros project)
--> Ready to merge from my end 👍 

One thing I'd like to propose to you is the option of providing a importable WSL2 distro like you provided the VM. Maybe this would make it much easier for new and probably mostly windows users to start working without the need for a powerful computer. My Intel Pentium Surface Go 1st gen is more than enough to run WSL2 really smooth.

Reply:
> One thing I'd like to propose to you is the option of providing a importable WSL2 distro like you provided the VM. Maybe this would make it much easier for new and probably mostly windows users to start working without the need for a powerful computer. My Intel Pentium Surface Go 1st gen is more than enough to run WSL2 really smooth.

In principle I like the idea, I am not sure how to implement it in practice though. I am creating a new tickets to track the idea and experiment a bit with it: #1149.